chia/consensus/block_body_validation.py

```diff
@@ -126,15 +121,12 @@ def rollback(self, header_hash: bytes32, height: int) -> None:
 
 async def validate_block_body(
     constants: ConsensusConstants,
-    blocks: BlockchainInterface,
-    block_store: BlockStore,
-    coin_store: CoinStore,
-    peak: Optional[BlockRecord],
+    get_block_record_from_db: Callable[[bytes32], Awaitable[Optional[BlockRecord]]],
```

Comment:
I think this might not be desirable. I think we want to move towards this function not returning an `Awaitable`, but not being async at all. Long term, possibly passing in the `BlockRecord`s potentially necessary for the validation